crates/bins/src/bin/datadog_static_analyzer_server/cli.rs

```diff
@@ -213,43 +211,46 @@ pub fn prepare_rocket() -> Result<RocketPreparation, CliError> {
+                    tracing::info!("Starting the keep alive loop");
 
-                tracing::info!("Starting the keep alive loop");
+                    loop {
+                        // get the latest request timestamp and the current one
+                        // remember that the `last_ping_request_timestamp_ms` state will be written by a fairing on every successful request
+                        let latest_timestamp = last_ping_request_timestamp_ms
+                            .try_read()
+                            .map(|x| *x)
+                            .unwrap_or_default();
 
-                loop {
-                    // get the latest request timestamp and the current one
-                    // remember that the `last_ping_request_timestamp_ms` state will be written by a fairing on every successful request
-                    let latest_timestamp = last_ping_request_timestamp_ms
-                        .try_read()
-                        .map(|x| *x)
-                        .unwrap_or_default();
+                        let current_timestamp = get_current_timestamp_ms();
 
-                    let current_timestamp = get_current_timestamp_ms();
-
-                    if latest_timestamp > 0 && current_timestamp > latest_timestamp + timeout_ms {
-                        eprintln!("Exiting because of timeout. Trying to exit gracefully");
-                        tracing::info!("Exiting because of timeout. Trying to exit gracefully");
-                        shutdown_handle.notify();
-                        // we give 10 seconds for the process to terminate
-                        // if it does not, we abort
-                        thread::sleep(Duration::from_secs(10));
-                        eprintln!("No graceful exit, aborting the process");
-                        tracing::error!("No graceful exit, aborting the process");
-                        // dropping the guard here to flush the pending tracing messages before exiting abruptly
-                        drop(cloned_guard);
-                        process::abort();
+                        if latest_timestamp > 0 && current_timestamp > latest_timestamp + timeout_ms
+                        {
+                            tracing::info!("Exiting because of timeout. Trying to exit gracefully");
+                            shutdown_handle.notify();
+                            // we give 10 seconds for the process to terminate
+                            // if it does not, we abort
+                            sleep(Duration::from_secs(10)).await;
+                            tracing::error!(
+                                "No graceful exit on first attempt. Trying another channel"
+                            );
+                            // signal the main thread to kill itself. wait for 10 secs and use abort if needed.
+                            let _ = tx_keep_alive_error.send(101).await;
```

Comment:
We should probably move the `101` here, `100` below, and other exit codes to a constant:

```rust
const ERROR_CHANNEL_DISCONNECTED: i32 = 100;
const ERROR_SHUTDOWN_FAILURE: i32 = 101;
const ERROR_UNKNOWN: i32 = 102;
const ERROR_BAD_ADDRESS: i32 = 14;
```

Reply:
Good suggestion! Done!